netmaster/k8snetwork/networkpolicy.go

```diff
@@ -391,8 +803,626 @@ func InitK8SServiceWatch(listenURL string, isLeader func() bool) error {
+		//build Ingress PodSelector obj
+		fromPodSelector, err := k8sNet.
+			getIngressPodSelectorList(policy.From, nameSpace)
+		if err != nil {
+			return []k8sIngress{}, err
+		}
+		//npLog.Infof("fromPodSelector:%+v", fromPodSelector)
+		ingressRules = append(ingressRules,
+			k8sIngress{IngressRules: rules,
+				IngressPodSelector: fromPodSelector})
+	}
+	return ingressRules, nil
+}
+
+func (k8sNet *k8sContext) getPodsIpsSetUsingLabel(m map[string]string,
+	nameSpace string) ([]string, error) {
+	var ipList []string
+	// labels.Parser
+	labelSectorStr := labels.SelectorFromSet(labels.Set(m)).String()
+	//Quary to K8s Api server for pods of given Label selector
+	podsList, err := k8sNet.k8sClientSet.CoreV1().
+		Pods(nameSpace).
+		List(meta_v1.ListOptions{LabelSelector: labelSectorStr})
+	if err != nil {
+		npLog.Fatalf("failed to get Pods from  K8S Server, %v", err)
+		return nil, err
+	}
+	for _, pod := range podsList.Items {
+		ipList = append(ipList, pod.Status.PodIP)
+	}
+	return ipList, nil
+}
+func (k8sNet *k8sContext) initPodSelectorCacheTbl(m map[string]string,
+	podSelector *k8sPodSelector) error {
+	if podSelector == nil {
+		return fmt.Errorf("Passe Nil Pod Selector reference")
+	}
+	//XXX:Don't confused PodSelector with Pod: PodSelector object keeps all
+	//attched label Ips
+	if len(podSelector.podIps) <= 0 {
+		podSelector.podIps = make(map[string]string, 0)
+		npLog.Infof("Init PodSelector podIp table:%v",
+			podSelector.labelPodMap)
+	}
+	//PodSelector: Keep track of all its label
+	if len(podSelector.labelPodMap) <= 0 {
+		podSelector.labelPodMap = make(map[string]map[string]bool, 0)
+		for key, val := range m {
+			lkey := getLabelSelector(key, val)
+			podSelector.labelPodMap[lkey] = make(map[string]bool, 0)
+		}
+		npLog.Infof("Init PodSelector Map table:%v",
+			podSelector.labelPodMap)
+	}
+	return nil
+}
+
+//Create podSelector object and Init its attributes i.e podIps , label etc
+func (k8sNet *k8sContext) parsePodSelectorInfo(m map[string]string,
+	nameSpace string) (*k8sPodSelector, error) {
+
+	PodSelector := k8sPodSelector{
+		TenantName:  getTenantInfo(),
+		NetworkName: getNetworkInfo(),
+		GroupName:   getEpgInfo()}
+
+	npLog.Infof("Build PodSelector Info using Label:%+v", m)
+
+	labelSectorStr := labels.SelectorFromSet(labels.Set(m)).String()
+
+	//Quary to K8s Api server using label Selector to get Pods list
+	podsList, err := k8sNet.k8sClientSet.CoreV1().
+		Pods(nameSpace).
+		List(meta_v1.ListOptions{LabelSelector: labelSectorStr})
+	if err != nil {
+		npLog.Fatalf("failed to get Pods from  K8S Server, %v", err)
+		return nil, err
+	}
+	if err = k8sNet.initPodSelectorCacheTbl(m, &PodSelector); err != nil {
+		return nil, err
+	}
+	//Build mapping for Label To PodIP
+	for _, pod := range podsList.Items {
+		for key, val := range pod.ObjectMeta.Labels {
+			lkey := getLabelSelector(key, val)
+			npLog.Infof("Update label Selector key:%v", lkey)
+			if ipMap, ok := PodSelector.labelPodMap[lkey]; ok {
```

Comment:
marking a spot that I think is aligning with OR based logic for label matching instead of AND